packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -439,3 +439,18 @@ function validateMaxSessionDuration(duration?: number) {
     throw new Error(`maxSessionDuration is set to ${duration}, but must be >= 3600sec (1hr) and <= 43200sec (12hrs)`);
   }
 }
+
+/**
+ * A PolicyStatement that doesn't normalize its Principal field.
```

Comment:
Sounds to me like it DOES normalize the principal field (it normalizes `"*"` to `{AWS:"*"}`, no?).